src/store/postgres/builder.rs

```diff
@@ -14,6 +15,27 @@ use crate::Aggregate;
 use super::persistable::Persistable;
 use super::{PgStore, Schema};
 
+/// The `EventIdUuidFormat` enum defines the UUID format of the event ID:
+///
+/// - `V4`: Uses the random UUID version 4 as defined by RFC 9562 section 5.4.
+/// - `V7`: Uses the time-ordered UUID version 7 as defined by RFC 9562 section 5.7.
+/// - `Custom`: Accepts a function used to generate UUIDs.
+pub enum EventIdUuidFormat {
+    V4,
+    V7,
+    Custom(fn() -> Uuid),
+}
+
+impl Into<fn() -> Uuid> for EventIdUuidFormat {
+    fn into(self) -> fn() -> Uuid {
+        match self {
+            EventIdUuidFormat::V4 => Uuid::new_v4,
+            EventIdUuidFormat::V7 => Uuid::now_v7,
```

Comment:
I like this. Maybe worth turning this into a `UuidFormat = V4 | V7` and allow the builder to customise both EventId and AggregateId **independently**?

Reply:
I'm not sure how that would work gives the builder does not initialize aggregates?

I could move `UuidFormat` so it's defined in src/state.rs and then define `AggregateState::with_id_format`. I don't need that myself, though, and I think you generally would want a random ID for an aggregate.

Reply:
You're right, the aggregate id is not controlled by the Store. This is fine!